pypsa_to_IPCC.py

```diff
@@ -121,136 +121,226 @@
             ds = file['data' + str(i)] 
             var={}
             
-            """
-            Capacity : Solar PV, onshore and offshore wind
-            """
-            #MW -> GW
-            var['Capacity|Electricity|Solar|PV'] =0.001*n.generators.p_nom_opt[country + ' solar'] if country + ' solar' in n.generators.index else 0
-            var['Capacity|Electricity|Solar'] = var['Capacity|Electricity|Solar|PV']
-            var['Capacity |Electricity|Solar|PV|Rooftop PV'] = 0.5 * var['Capacity|Electricity|Solar|PV']
-            var['Capacity|Electricity|Solar|PV|Utility-scale PV'] = 0.5 * var['Capacity|Electricity|Solar|PV']
-            if country in wind_split:
-                var['Capacity|Electricity|Wind|Onshore']=0.001*n.generators.p_nom_opt[[i for i in n.generators.index if country in i and 'onwind' in i]].sum() 
-                var['Capacity|Electricity|Wind|Offshore']=0.001*n.generators.p_nom_opt[[i for i in n.generators.index if country in i and 'offwind' in i]].sum() 
-            else:
-                var['Capacity|Electricity|Wind|Onshore']=0.001*n.generators.p_nom_opt[country + ' onwind'] if country + ' onwind' in n.generators.index else 0
-                var['Capacity|Electricity|Wind|Offshore']=0.001*n.generators.p_nom_opt[country + ' offwind'] if country + ' offwind' in n.generators.index else 0
-            var['Capacity|Electricity|Wind']=var['Capacity|Electricity|Wind|Onshore']+var['Capacity|Electricity|Wind|Offshore']
+            dict_var={'Capacity':' ', 'Capacity Additions':year}      
+            for v_type in dict_var.keys():
+               """
+               Capacity : Solar PV, onshore and offshore wind
+               """
+               #MW -> GW
+               var[v_type+'|Electricity|Solar|PV'] =0.001*(n.generators.p_nom_opt.filter(like ='solar').filter(like =country).filter(like=str(dict_var[v_type])).sum()-
+                                                          n.generators.p_nom_opt.filter(like ='solar thermal').filter(like =country).filter(like=str(dict_var[v_type])).sum())
```

Comment:
this is a little counterintuitive, I suggest adding a comment saying #solar thermal is subtracted to only account for solar PV

Reply:
Good point, I added a comment.

Reply:
thanks